pubsub-mapped-api/src/test/java/com/google/pubsub/clients/producer/ProducerTest.java

```diff
@@ -0,0 +1,215 @@
+import org.apache.kafka.common.serialization.Deserializer;
+import org.apache.kafka.common.serialization.IntegerSerializer;
+import org.apache.kafka.common.serialization.StringDeserializer;
+import org.apache.kafka.common.serialization.IntegerDeserializer;
+
+import java.io.IOException;
+import java.util.Properties;
+import com.google.api.core.ApiFuture;
+
+import org.mockito.Mockito;
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.modules.junit4.PowerMockRunner;
+import static org.powermock.api.mockito.PowerMockito.when;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+
+/**
+ * Created by gewillovic on 8/1/17.
+ */
+
+@RunWith(PowerMockRunner.class)
+@PrepareForTest({Publisher.class, TopicAdminClient.class})
+public class ProducerTest {
+
+  private ApiFuture lf;
+  private Publisher stub;
+  private Builder stubBuilder;
+  private ProducerConfig config;
+  private Serializer serializer;
+  private Deserializer deserializer;
+  private TopicAdminClient topicAdmin;
+  private ArgumentCaptor<PubsubMessage> captor;
+  private KafkaProducer<String, Integer> publisher;
+
+  @Before
+  public void setUp() throws IOException {
+    Properties properties = new Properties();
+    properties.putAll(new ImmutableMap.Builder<>()
+        .put("acks", "0")
+        .put("project", "project")
+        .put("key.serializer", "org.apache.kafka.common.serialization.StringSerializer")
+        .put("value.serializer", "org.apache.kafka.common.serialization.IntegerSerializer")
+        .build()
+    );
+
+    lf = PowerMockito.mock(ApiFuture.class);
+    captor = ArgumentCaptor.forClass(PubsubMessage.class);
+
+    stub = PowerMockito.mock(Publisher.class, Mockito.RETURNS_DEEP_STUBS);
+    stubBuilder = PowerMockito.mock(Builder.class, Mockito.RETURNS_DEEP_STUBS);
+    topicAdmin = PowerMockito.mock(TopicAdminClient.class, Mockito.RETURNS_DEEP_STUBS);
+
+    config = new ProducerConfig(
+        ProducerConfig.addSerializerToConfig(properties, null, null));
+
+    PowerMockito.mockStatic(Publisher.class);
+    PowerMockito.mockStatic(TopicAdminClient.class);
+
+    when(stubBuilder.build()).
+        thenReturn(stub);
+    when(TopicAdminClient.create()).
+        thenReturn(topicAdmin);
+    when(stub.publish(captor.capture())).
+        thenReturn(lf);
+    when(Publisher.defaultBuilder(Matchers.<TopicName>any())).
+        thenReturn(stubBuilder);
+    when(stubBuilder.setRetrySettings(Matchers.<RetrySettings>any())).
+        thenReturn(stubBuilder);
+    when(stubBuilder.setBatchingSettings(Matchers.<BatchingSettings>any())).
+        thenReturn(stubBuilder);
+
+    publisher = new KafkaProducer<String, Integer>(config, null, null);
+  }
+
+  @Test
+  public void testFlush() {
+    publisher.send(new ProducerRecord<String, Integer>("topic", 123));
+
+    publisher.flush();
+
+    publisher.send(new ProducerRecord<String, Integer>("topic", 456));
+
+    try {
+      Mockito.verify(stub, Mockito.times(1)).
```

Comment:
I believe that many of these Mockito.verify... commands will fit on single lines.